app/Services/ProvisioningService.php

```diff
@@ -0,0 +1,366 @@
+
+        return ltrim($name);
+    }
+
+    protected function createWrapper(object $properties, callable $callback)
+    {
+        Log::notice("Provisioning {$this->modelName()} '{$this->instanceName($properties)}'");
+        $validator = Validator::make((array) $properties, $this->expectedProperties);
+        if ($validator->fails()) {
+            throw new UnexpectedValueException("Invalid {$this->modelName()} definition: {$validator->errors()}");
+        }
+        $item = new $this->model;
+        $callback($item);
+        $item->save();
+    }
+
+    protected function destroyWrapper(array $match, ?callable $callback = null)
+    {
+        if ($match) {
+            $expression = implode(' && ', array_map(fn ($a, $b) => "$a = $b", array_keys($match), array_values($match)));
+            Log::notice("Deleting all {$this->modelName()}s matching '$expression'");
+        } else {
+            Log::notice("Deleting all {$this->modelName()}s");
+        }
+        $items = $this->model::lazy();
+        foreach ($match as $key => $value) {
+            $items = $items->where($key, $value);
+        }
+        foreach ($items as $item) {
+            if ($callback) {
+                $callback($item);
+            }
+            if (! $item->delete()) {
+                Log::error("Failed to delete {$this->modelName()} '{$item->getLogLabel()}'");
+            }
+        }
+    }
+
+    abstract public function create(object $properties);
+
+    // abstract public function read(array $match): object;
+
+    // abstract public function update(array $match, object $properties);
+
+    abstract public function destroy(array $match = []);
+}
+
+class ServerProvisioner extends AbstractProvisioner
+{
+    protected string $model = Server::class;
+
+    protected array $expectedProperties = [
+        'name' => 'required|string|unique:servers,name',
+        'description' => 'required|string',
+        'endpoint' => 'required|string',
+        'secret' => 'required|string',
+        'strength' => 'required|integer|min:1|max:10',
+        'status' => 'required|in:disabled,enabled',
+    ];
+
+    public function create(object $properties)
+    {
+        $this->createWrapper($properties, function ($srv) use ($properties) {
+            // TODO: PHP 8.3 allows the following syntax
+            // ServerStatus::{strtoupper($properties->status)}->value;
+            $status = "App\Enums\ServerStatus::".strtoupper($properties->status);
+            if (! defined($status)) {
+                throw new UnexpectedValueException('Invalid server status');
+            }
+            $srv->name = $properties->name;
+            $srv->description = $properties->description;
+            $srv->base_url = $properties->endpoint;
+            $srv->secret = $properties->secret;
+            $srv->strength = $properties->strength;
+            $srv->status = constant($status)->value;
```

Comment:
```suggestion
            try{
                $status = App\Enums\ServerStatus::{strtoupper($properties->status)};
            }
            catch (Error $e){
                throw new UnexpectedValueException('Invalid server status');
            }
            
            $srv->name = $properties->name;
            $srv->description = $properties->description;
            $srv->base_url = $properties->endpoint;
            $srv->secret = $properties->secret;
            $srv->strength = $properties->strength;
            $srv->status = $status;
```

Reply:
Thanks, I couldn't get this to work. :pray: 
`try / catch` is unnecessary because of input validations.